### PR TITLE
Upgrading Actix

### PR DESCRIPTION
**Pull request by knabben**
Moving actix to latest 0.8.3 and actix-web to 1.0.0

Related to: https://jira.hyperledger.org/browse/GRID-78

### CODE REVIEW

**Comment by jsmitchell** [comment]
Hi @knabben - thanks for the contribution! One of your commits is missing signoff. You can fix that branch by doing `git commit --amend --signoff` and then force pushing.

**Comment by peterschwarz** [comment]
Could you also update your commit messages to follow: https://chris.beams.io/posts/git-commit/  Specifically your second commit gives no context to the change (I thought it was about the DB migrations, and it sounded totally unrelated, for example).

**Comment by knabben** [comment]
sure, after figuring out what is breaking at compilation time, I do both action. Gonna pay more attention on PR commits, in the mean time can some one run this on Jenkins or add my user to the whitelist, so I can see which tests are breaking as well.

File: daemon/src/rest_api/routes/batches.rs [hunk 1]
```diff
@@ -159,9 +159,9 @@ impl Handler<BatchStatuses> for SawtoothMessageSender {
 }
 
 pub fn submit_batches(
-    (req, state): (HttpRequest<AppState>, State<AppState>),
+    (req, state, body): (HttpRequest, web::Data<AppState>, web::Bytes),
 ) -> impl Future<Item = HttpResponse, Error = RestApiResponseError> {
-    req.body().from_err().and_then(
+    body.from_err().and_then(
```
Comment (knabben) [inline daemon/src/rest_api/routes/batches.rs]: This is failing with: 
```
error[E0599]: no method named `from_err` found for type `bytes::bytes::Bytes` in the current scope
   --> src/rest_api/routes/batches.rs:164:10
    |
164 |     body.from_err().and_then(
    |          ^^^^^^^^
    |
    = note: the method `from_err` exists but the following trait bounds were not satisfied:
            `&mut [u8] : futures::future::Future`
            `&mut bytes::bytes::Bytes : futures::future::Future`
```

**Comment by peterschwarz** [comment]
Would you mind taking this issue in Jira: https://jira.hyperledger.org/browse/GRID-78

**Comment by peterschwarz** [comment]
Looks like you have lint - run `cargo fmt` and fixup the appropriate commits

File: daemon/src/rest_api/routes/batches.rs (hunk 1 above)
Comment (peterschwarz) [reply daemon/src/rest_api/routes/batches.rs]: it looks like you don't need that `body.from_err().and_then(...`, you can just use the body directly and remove the wrapping function.  Once that is corrected, a whole new host of errors crop up, though.

Comment (knabben) [reply daemon/src/rest_api/routes/batches.rs]: True, I need more research on Rust to come up with the final solution, closing this for now. The guide: https://github.com/actix/actix-web/blob/master/MIGRATION.md is a good start.

**Pull request closed without merging**